feat: Add Docker containerization and comprehensive contributing guide

KaranNegi20Feb opened the pull request:
## Issue Reference

Closes #258

## Changes

Added docker-compose.yml for containerized development
Created start.sh script for automated Docker environment setup
Added pnpm run start:docker command to package.json
Added comprehensive CONTRIBUTING.md with development guidelines

## Usage
pnpm run start:docker

# Existing local workflow still works
pnpm run dev

KaranNegi20Feb commented:
hey @idanlodzki make sure you check the contribution.md i made it a little better with some new stuff

yuvalmich commented on start.sh:
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
```
please create scripts directory and put it inside.

In addition, I would rename it to start-docker.sh

KaranNegi20Feb replied:
yeah that's something idan said too 

I'll make sure both of your suggestions are implemented, just give me 2 hours.

yuvalmich reviewed (commented):
Looks Good!

KaranNegi20Feb commented:
@yuvalmich  i have changed the name from start to run 
but on your recommendation i have added them to a dir called scripts
 @yuvalmich did this now
 "run.sh" that the user can do command like: curl https://raw.githubusercontent.com/opsimate/opsimate/scripts/start.sh | sh

KaranNegi20Feb commented:
Changed the name to
 https://raw.githubusercontent.com/OpsiMate/OpsiMate/main/scripts/start-docker.sh | sh
 start-docker.sh
make it more explicit

you can also test it via 
sh scripts/start-docker.sh  from the repo using any of the general IDE's terminal you may prefer
